src/powerbox/powerbox.py

```diff
@@ -141,14 +142,18 @@ def __init__(
         seed=None,
         nthreads=None,
     ):
-        self.N = N
         self.dim = dim
+        self.N = N
+        if isinstance(self.N, numbers.Number):
+            self.N = [self.N] * self.dim
         self.boxlength = boxlength
+        if isinstance(self.boxlength, numbers.Number):
+            self.boxlength = [self.boxlength] * self.dim
```

Comment:
Somewhere in the `__init__` there should be some checks that the length of `N` and `boxsize` are the same, and equivalent to `dim`.